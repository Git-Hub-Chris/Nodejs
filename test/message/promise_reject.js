'use strict';
const common = require('../common');

// Flags: --expose-gc

Promise.reject(new Error('oops'));

// Manually call GC due to possible memory contraints with attempting to
// trigger it "naturally".
setTimeout(common.mustCall(() => {
  /* eslint-disable no-undef */
  gc();
  gc();
  gc();
  /* eslint-enable no-undef */
}, 1), 2);

process.on('beforeExit', () =>
    common.fail('beforeExit should not be reached'));

process.on('uncaughtException', (err) => {
    // XXX(Fishrock123): This test is currently broken...
    console.log(err.stack);
    common.fail('Should not trigger uncaught exception');
});

process.on('exit', () => process._rawDebug('exit event emitted'));
