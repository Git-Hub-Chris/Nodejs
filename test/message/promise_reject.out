exit event emitted
*test*message*promise_reject.js:*
Promise.reject(new Error('oops'));
               ^

Error: oops
    at *test*message*promise_reject.js:*:*
    at Module._compile (module.js:*:*)
    at Object.Module._extensions..js (module.js:*:*)
    at Module.load (module.js:*:*)
    at tryModuleLoad (module.js:*:*)
    at Function.Module._load (module.js:*:*)
    at Module.runMain (module.js:*:*)
    at run (bootstrap_node.js:*:*)
    at startup (bootstrap_node.js:*:*)
    at bootstrap_node.js:*:*
